src/vs/workbench/contrib/bulkEdit/browser/bulkEditService.ts

```diff
@@ -239,6 +264,13 @@ export class BulkEditService implements IBulkEditService {
 			!!options?.confirmBeforeUndo
 		);
 
+		for (const edit of edits) {
+			if (edit instanceof ResourceAttachmentEdit) {
+				const widget = this._chatWidgetService.getWidgetByInputUri(edit.resource);
+				widget?.attachmentModel.addContext(edit.variable);
+			}
+		}
```

Comment:
This doesn't seem right. The edit service should not know about the chat widget service but only about opaque edits that do something and can be undone. This mean, no dependeny here, no such logic, but pass an OpaqueEdit which (in this one instance) does/undoes this.